samples/cpp_synchronization/sample.yaml

```diff
@@ -1,7 +1,7 @@
 sample:
   name: Syncronisation (C++)
 tests:
-  sample.cpp_syncronisation:
+  sample.cpp.syncronisation:
```

Comment:
This is just one example; also all files in `samples/drivers` became `sample.peripheral`.

Reply:
I changed peripheral to drivers, this should align with the `include/drivers`, `drivers/` and `samples/drivers` and `tests/drivers` however, we want to decouple the test identifier from the path, in most cases this will match perfectly, especially where we have good structure already or where samples or tests are places in the right location, however based on the experience from the past, tests and samples move around and we want to keep the identifier independent of where the tests are maintained. In this case cpp is the category (main section in the testsuite)

Reply:
OK.  It's a pain to search for the tests when they're not in the location implied by the identifier, but I understand name stability may be more important than locatability.